packages/grid/x-data-grid-premium/src/DataGridPremium/useDataGridPremiumComponent.tsx

```diff
@@ -60,6 +60,8 @@ import {
   columnGroupsStateInitializer,
```

Comment:
@MBilalShafi 
> we would require to apply filtering and aggregation in some use cases, for example when we lazy-load row children, we will need to reapply filtering and aggregation to re-validate the stale data.

I'm not sure about this...
Do client-side filtering and aggregation make sense when lazy loading group children from the server?
Technically, it could work, but aggregating and filtering only a portion of the data on the client doesn't seem very useful.

Reply:
That's a good question, and I feel we are not on the same page on this.

How I want the lazy-load properties implementation to work is, the grid will work in Hybrid mode, i.e. users can provide initial `treeData` on the client side and they can provide two extra props using which they can lazy-load row children for certain rows, but they can still use methods like `setRows`, `updateRows` and even update `rows` prop on the client side, here are the two extra props as suggested in the original ticket:
```js
hasChildrenOnServer?: (row: GridRowModel) => boolean;
fetchRowChildren?: (row: GridRowModel) => Promise<GridRowModel[]>;
```

Now, some rows could be client side and some could be server-side (and even hybrid 🤔 ). As this is not the final solution, a full fledge solution would require to have a data fetching layer like AGGrid has ([context](https://github.com/mui/mui-x/issues/210#issuecomment-988589878)) and then we can define it something like `treeDataMode="server"` to fully make it server side.

As this one is a quick solution that unlocks people to achieve something closer to the full fledge solution, I wanted to have all the other features like filtering, sorting, aggregation, etc work in the same way as on the client side, it's the same as if a user is doing `apiRef.current.updateRows` on the client side. Filtering, aggregation, etc have to work although new rows are added. For example, on this [demo](https://codesandbox.io/s/nifty-glade-36nexh?file=/demo.tsx) filtering persists on new rows adding.

Users can move filtering, sorting, etc to the server-side using `filteringMode='server'` for example, but it may not work ideally as we are only updating part of rows using `apiRef.current.updateRows` instead of updating all the rows for performance reasons.

Let me know if you have a different understanding.

Reply:
I just realized that when lazy loading group children, the filters should be applied **after** the new rows are added to the grid, not on group expansion 🙂
Do you see any other use cases for applying filters on `rowExpansionChange`?

Reply:
That makes sense 👍